server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -954,24 +999,32 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
       // delete any entries for external compactions
       extCompIds.forEach(ecid -> m.putDelete(ExternalCompactionColumnFamily.STR_NAME, ecid));
 
-      if (!m.getUpdates().isEmpty()) {
-        bw.addMutation(m);
-      }
-
-      bw.flush();
+      // Add a marker so we know the tablets have been fenced in case the merge operation
+      // needs to be recovered and restarted to finish later. The value is firstPrevRowValue
+      // which we need if we need to restart the operation
+      MergedColumnFamily.MERGED_COLUMN.put(m,
+          MergedColumnFamily.encodeHasPrevRowColumn(firstPrevRowValue != null));
```

Comment:
Can firstPrevRowValue ever be non null at this point?  Bit uncertain about this, thinking about the following when I ask this question.

If when merge operation starts there is only a single tablet in the merge range, then it will never call `mergeMetadataRecords()`.  So the first time that `mergeMetadataRecords()` is called it should always see at least two tablets and I think `firstPrevRowValue` is always non null when there are at least two tablets.   The first update made by `mergeMetadataRecords()` is done before any tablets are deleted and adds the new merged marker. If the above is true, then it seems when the merge marker is successfully added that it will have seen at least two tablets and  `firstPrevRowValue` will be non null?

Reply:
I think you are right that this wouldn't be null so we can get rid of this and will make things much simpler as we just need to worry about the merge marker existing.